Findovsrpro.cmake

```diff
@@ -1,132 +1,111 @@
-    endif()
-    string(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)?"
-      "gcc${digits}"
-      prefix ${GCC_VERSION}
-      )
-  elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang") # LLVM/Apple Clang (clang.llvm.org)
-    if(${CMAKE_SYSTEM_NAME} STREQUAL Darwin)
-      exec_program(${CMAKE_CXX_COMPILER}
-        ARGS ${CMAKE_CXX_COMPILER_ARG1} -dumpversion
-        OUTPUT_VARIABLE CLANG_VERSION
-        )
-      string(REGEX REPLACE "([0-9]+)\\.([0-9]+)(\\.[0-9]+)?"
-        "clang-darwin\\1\\2" # match boost naming
-        prefix ${CLANG_VERSION}
-        )
-    else()
-      string(REGEX REPLACE "([0-9]+)\\.([0-9]+)(\\.[0-9]+)?"
-        "clang\\1\\2" # match boost naming
-        prefix ${CMAKE_CXX_COMPILER_VERSION}
-        )
-    endif()
-  else()
-    message(SEND_ERROR "Findovsrpro.cmake: compiler support lacking: ${CMAKE_CXX_COMPILER_ID}")
-  endif()
-  set(${_ret} ${prefix} PARENT_SCOPE)
-endfunction()
-function(getNumBits _ret)
-  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
-    set(numBits 64)
-  elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
-    set(numBits 32)
-  else()
-    message(FATAL_ERROR "numBits not 64 or 32")
+if(NOT ovsrpro_FIND_VERSION)
+  message(FATAL_ERROR "A version of Ovsrpro is required in the find_package() command.")
+endif()
+
+function(ovsrpro_message)
+  if(NOT ovsrpro_FIND_QUIETLY)
+    message(${ARGN})
   endif()
-  set(${_ret} ${numBits} PARENT_SCOPE)
 endfunction()
-################################################################################
-# TRICKY: clear cached variables each time we cmake so we can change
-# ovsrpro_REV and reuse the same build directory
-unset(ovsrpro_DIR CACHE)
-################################################################################
-# find the path to the ovsrpro directory
-getCompilerPrefix(COMPILER)
-getNumBits(BITS)
-set(ovsrpro_SIG ${ovsrpro_REV}-${COMPILER}-${BITS})
-set(PFX86 "ProgramFiles(x86)")
-find_path(ovsrpro_DIR
-  NAMES
-    ovsrpro_${ovsrpro_SIG}.txt
+
+if(ovsrpro_FIND_COMPONENTS)
+  ovsrpro_message(
+    AUTHOR_WARNING
+    "Ovsrpro does not support components in find_package()."
+  )
+endif()
+
+# Since ovsrpro uses date versioning instead of semantic versioning, we cannot
+# deduce compatibility from the version. So force an exact match to be found.
+if(NOT ovsrpro_FIND_VERSION_EXACT)
+  set(ovsrpro_FIND_VERSION_EXACT true)
+  ovsrpro_message(
+    AUTHOR_WARNING
+    "Ovsrpro requires an exact version match. We recomment adding EXACT to find_package()."
+  )
+endif()
+
+# Figure out the compiler name and version to search for.
+if(CMAKE_COMPILER_IS_GNUCXX)
+  execute_process(
+    COMMAND
+      ${CMAKE_CXX_COMPILER}
+      ${CMAKE_CXX_COMPILER_ARG1}
+      -dumpfullversion
+      -dumpversion
+    OUTPUT_VARIABLE gcc_version
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  string(
+    REGEX REPLACE
+    "([0-9]+)\\.([0-9]+)\\.([0-9]+)?"
+    "gcc\\1\\2\\3"
+    compiler
+    ${gcc_version}
+  )
+else()
+  message(FATAL_ERROR "${CMAKE_CXX_COMPILER_ID} is not supported.")
```

Comment:
Not allowing Windows builds now?  I know we only ever kindof supported them, not attempting to support them any longer?

Reply:
Not until some downstream project needs Windows support.